xayn-ai/src/coi/system.rs

```diff
@@ -236,8 +240,8 @@ fn log_negative_user_reaction(
     embedding: &Embedding,
     config: &Configuration,
 ) {
-    match find_closest_coi_mut(cois, embedding, config.neighbors()) {
-        Some((coi, distance)) if distance < config.threshold() => {
+    match find_closest_coi_mut(cois, embedding) {
+        Some((coi, distance)) if distance >= config.threshold() => {
```

Comment:
it's still called `distance` instead of `similarity`
```suggestion
        Some((coi, similarity)) if similarity >= config.threshold() => {
```